### ISSUE TITLE
Exposure Check Failed

### ISSUE THREAD

**Issue by lumpiluk** [corona-warn-app/cwa-app-android#4740]
## Avoid duplicates
* [x] Bug is not mentioned in the [FAQ](https://www.coronawarn.app/en/faq/)
* [X] Bug is specific for Android only, for general issues / questions that apply to iOS and Android please raise them in the [documentation repository](https://github.com/corona-warn-app/cwa-documentation)
* [X] Bug is not already reported in another issue

There is a related issue in the FAQ: "Exposure Check Failed after updating to version 1.6.0", but I'm using 2.16.2.

## Technical details

- Device name: Nokia 7 plus
- Android version: 10
- App version: 2.16.2

## Describe the bug

For the 3rd or 4th day in a row I only get this message: "Exposure check failed" – "The synchronization of random IDs with the server failed. You can restart the synchronization manually." Tapping "Restart" shortly shows "Data is being downloaded" and "Check running", but then the status remains as before.

## Steps to reproduce the issue

Open the app and check the status page.

## Expected behaviour

I know that my status should be 'red' from a check-in last week.

## Possible Fix

-

## Additional context

Maybe this excerpt from the log helps? The date and time with high risk would match the event I checked into.

Possibly relevant: I forgot to check in when I entered the event, so I did it afterwards when I left and immediately checked out again. All people were still in the room (wearing masks, by the way), so I hadn't bothered updating the start time of the event until a few days ago when my status was either still green or already showed the error – I don't remember, unfortunately.

```
LastCombinedRiskResults(lastCalculated=CombinedEwPtRiskLevelResult(ptRiskLevelResult=PtRiskLevelResult(calculatedAt=2022-01-25T06:40:25.339Z, riskState=CALCULATION_FAILED, presenceTracingDayRisk=[PresenceTracingDayRisk(localDateUtc=2022-01-19, riskState=CALCULATION_FAILED)], checkInWarningOverlaps=[CheckInWarningOverlap(checkInId=31, transmissionRiskLevel=6, traceWarningPackageId=DE_456377, startTime=2022-01-19T19:05:17.458Z, endTime=2022-01-19T19:05:30.923Z)]), ewRiskLevelResult=RiskLevelTaskResult(calculatedAt=2022-01-25T06:40:28.179Z, failureReason=null, ewAggregatedRiskResult=EwAggregatedRiskResult(totalRiskLevel=HIGH, totalMinimumDistinctEncountersWithLowRisk=4, totalMinimumDistinctEncountersWithHighRisk=1, mostRecentDateWithLowRisk=2022-01-17T00:00:00.000Z, mostRecentDateWithHighRisk=2022-01-19T00:00:00.000Z, numberOfDaysWithLowRisk=1, numberOfDaysWithHighRisk=1, exposureWindowDayRisks=null), exposureWindows.size=null), exposureWindowDayRisks=[ExposureWindowDayRisk(dateMillisSinceEpoch=1642550400000, riskLevel=HIGH, minimumDistinctEncountersWithLowRisk=1, minimumDistinctEncountersWithHighRisk=1), ExposureWindowDayRisk(dateMillisSinceEpoch=1642377600000, riskLevel=LOW, minimumDistinctEncountersWithLowRisk=3, minimumDistinctEncountersWithHighRisk=0)]), lastSuccessfullyCalculated=CombinedEwPtRiskLevelResult(ptRiskLevelResult=PtRiskLevelResult(calculatedAt=2022-01-23T18:02:10.534Z, riskState=LOW_RISK, presenceTracingDayRisk=null, checkInWarningOverlaps=null),
```

---
Internal Tracking ID: [EXPOSUREAPP-11583](https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-11583)

**Comment by Ein-Tim** [comment]
Is the app also showing "Exposure logging stopped"?

**Comment by lumpiluk** [comment]
> Is the app also showing "Exposure logging stopped"?

No, at the top of the screen it reads "Exposure logging active"

**Comment by MikeMcC399** [comment]
@lumpiluk 

Has your device been set to a future date and then reset back to the current date?

**Comment by larswmh** [comment]
Thanks for your report @lumpiluk. We have created an internal ticket for it and will raise this topic internally.
Internal Tracking ID: [EXPOSUREAPP-11583](https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-11583)

---
Corona-Warn-App Open Source Team

**Comment by lumpiluk** [comment]
> @lumpiluk
> 
> Has your device been set to a future date and then reset back to the current date?

Not to my knowledge

**Comment by vaubaehn** [comment]
@lumpiluk 
For me it sounds like an issue with downloading DiagnosisKeys.
Possible reasons:
- problem with network (did you try wifi and mobile?)
- problem with free disk storage space (how much do you have left on internal memory)
- problem with timeout due to general system load

@MikeMcC399 do you have by chance an idea how big (bytes) are the DK packages (daily, hourly) currently? I can't check myself at the moment

**Comment by vaubaehn** [comment]
@lumpiluk as you already copied a line from the error report, you could try to find a failing download task. Unfortunately I don't have a good keyword to search for... I'd try `DownloadTask` or `DownloadDiagnosisKeys` (not sure whether these exist) or just `download` which will probably show up many hits in search.
If you find something that could point to the root cause.
The line you copied above only reflects some status values, the failure occurred somewhere before.

**Comment by lumpiluk** [comment]
I submitted the log using the App Information / Error Reports menu yesterday, where I also see a recommendation not to publish my log. I can find the listed keywords in the log and the only issue I can see with my lack of CWA experience is the line I posted above which appears after a `V/TracingStateProvider: riskLevelResults:`.
There is also a shorter version like this: `D/TracingStateProvider: TracingStateProvider FLOW emission: TracingInProgress(riskState=CALCULATION_FAILED, isInDetailsMode=false, tracingProgress=TracingProgress.IsCalculating)`

I doubt the network connection or system load are the problem as I've tried it on separate days both with Wi-Fi and the cellular connection and also after a reboot. Android reports more than 10 GB of free disk space.

Is there another way to submit my log and link it to this issue?

**Comment by MikeMcC399** [comment]
@lumpiluk 
> Is there another way to submit my log and link it to this issue?

If you submit an error report directly in the app then you can quote the ID here in this GitHub issue (see https://www.coronawarn.app/en/faq/#error_log). Since only the Open Source Team can access the reports this is only helpful if somebody on the Open Source Team is ready to pick up the error report and look at it.

Community members don't have access to error reports.

**Comment by lumpiluk** [comment]
Thank you, found it! This is the ID: B909160E7EEB6F3DD1EF

**Comment by lumpiluk** [comment]
> Yes it looks like it is failing because of the event checking risk calculation. Can you provide more details of the state of your checkins?

I currently have two check-ins in total, the likely relevant one on 2022-01-19 from 18:00 to 20:05 (after adjusting the duration as mentioned above). In the contact journal I see this for the same date:

- "Increased Risk based on the encounters evaluated by the app. They are not necessarily related to the people and places you have recorded."
- "Low Risk based on their presence at: […]"
- "Rapid Test Performed", "Negative Diagnosis"
- and the check-in itself, "02:00 hrs" (not precisely correct…)

And another "Low Risk" encounter on 2022-01-17.

**Comment by mtwalli** [comment]
This issue should be also fixed by https://github.com/corona-warn-app/cwa-app-android/pull/5411. Failed Presence Tracing calculation won't affect the aggregated risk status anymore

**Comment by MikeMcC399** [comment]
@lumpiluk 

- The fix https://github.com/corona-warn-app/cwa-app-android/pull/5411 which was released with [CWA Android 2.26](https://github.com/corona-warn-app/cwa-app-android/releases/tag/v2.26.0) is supposed to have fixed this issue (see https://github.com/corona-warn-app/cwa-app-android/issues/4740#issuecomment-1209081167)
- If you have updated to CWA 2.26 and you are no longer seeing the issue, then I suggest to close it here. CWA 2.26 was [released](https://www.coronawarn.app/en/blog/2022-08-31-cwa-2-26/) on Aug 31, 2022.
